src/NDItk/multigroup/Metadata/src/generateSecondaryGroups.hpp

```diff
@@ -0,0 +1,14 @@
+/**
+ *  @brief A helper function to generate the secondary group recors
+ */
+static std::vector< base::SingleIntegerRecord >
+generateSecondaryGroups( const std::map< unsigned int, unsigned int >& outgoing ) {
```

Comment:
I need an explanation here. Does this just get a particle - energy group pair, or does it add that to an existing map of particle-#ofgroups?

Reply:
When generating the num_grps_x keywords from scratch, the user give a map of particle type and number of group pairs. This function transforms that map into a vector of keywords, one num_grps for each particle type. Since the input is a map, I know that these are sorted as well.